Add functionality to write fasta with protein sequences

christopher-mohr opened the pull request:
Add functionality to write out a FASTA file with protein sequences as mentioned in #9 

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/epitopeprediction/tree/master/.github/CONTRIBUTING.md

skrakau reviewed (changes requested):
Hi, great that you added this! It is addressing https://github.com/nf-core/epitopeprediction/issues/9 (not https://github.com/nf-core/epitopeprediction/pull/33, although both dealing with FASTAs). I added a few comments.

Could you additionally add a description to the `docs/output.md`? 

And if I see it right, this option only works (and makes sense) with variants as input, i.e. not with `--peptides` or `--proteins`. Maybe it would be useful to check at the beginning of the pipline if the parameter combination is valid and throw an error if not. Also this could be mentioned in the help/schema description of this parameter.

christopher-mohr commented:
> Hi, great that you added this! It is addressing #9 (not #33, although both dealing with FASTAs). I added a few comments.

Right, changed it! 👍

christopher-mohr commented:
Thanks for the review @skrakau! Hope you're happy with the changes :)

skrakau commented on docs/output.md:
````diff
@@ -33,6 +33,8 @@ HTHVYLFLS 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpe
 HVYLFLSNL 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpeithi-1.0 0.0 0.0 False False False c.173C>A p.Pro58His
 ```
 
+When the parameter `--fasta_output` is specified a `FASTA` file will be generated that contains the sequences of proteins that are affected by the provided genomic variants. The resulting `FASTA` file will contain the wild-type and mutated protein sequences.
````
In which folder will this be located, and how will it be called?

skrakau reviewed (changes requested):
Looks good, just one tiny detail is still missing ;)

christopher-mohr commented:
The last commit includes more changes than initially planned ;):

- merge FASTA files in case of splitted results 
- fixed the statistics for single and multiple (splitter) results
- include input base name in result files
- change result folder name
- update docs

skrakau commented on docs/output.md:
```diff
@@ -16,11 +16,11 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
+**Output directory: `predictions/`**
 
-* `prediction_report.json`
+* `${input_base_name}_prediction_report.json`
```
This is inconsistent with the notation used for `[tool].[version].supported_alleles.txt`. Seems every pipeline does this differently. Use either `[...]` or `${...}` in the whole `output.md`.

skrakau commented on docs/output.md:
```diff
@@ -16,11 +16,11 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
+* `${input_base_name}_prediction_report.json`
   * The predicted epitopes in JSON format for downstream analysis tasks
-* `prediction_report.tsv`
+* `${input_base_name}_prediction_results.tsv`
```
See above.

skrakau commented on docs/output.md:
```diff
@@ -33,6 +33,13 @@ HTHVYLFLS 9 17 3336962 ENSG00000127780 ENST00000248384 ENSP00000248384 SNP syfpe
+
+**Output directory: `predictions/`**
+
+* `${input_base_name}_prediction_proteins.fasta`
```
See above.

skrakau reviewed (changes requested):
Looks good! :) 

Just one remaining inconsistency in the notation of output files.

skrakau reviewed (approved):
LGTM @christopher-mohr

christopher-mohr commented:
> LGTM @christopher-mohr

Thank you @skrakau !
